core/Plugin.php

```diff
@@ -461,6 +463,42 @@ public function getMissingDependenciesAsString($piwikVersion = null)
         ));
     }
 
+    /**
+     * Schedules re-archiving of this plugin's reports from when this plugin was last
+     * deactivated to now. If the last time core:archive was run is earlier than the
+     * plugin's last deactivation time, then we use that time instead.
+     *
+     * Note: this only works for CLI archiving setups.
+     *
+     * Note: the time frame is limited by the `[General] rearchive_reports_in_past_last_n_months`
+     * INI config value.
+     *
+     * @throws \DI\DependencyException
+     * @throws \DI\NotFoundException
+     */
+    public function schedulePluginReArchiving()
+    {
+        $lastDeactivationTime = $this->getPluginLastDeactivationTime();
+
+        $dateTime = null;
+
+        $lastCronArchiveTime = (int) Option::get(CronArchive::OPTION_ARCHIVING_FINISHED_TS);
+        if (empty($lastCronArchiveTime)) {
+            $dateTime = $lastDeactivationTime;
+        } else if (empty($lastDeactivationTime)) {
+            $dateTime = Date::factory($lastCronArchiveTime);
+        } else {
+            $lastCronArchiveTime = Date::factory($lastCronArchiveTime);
+            $dateTime = $lastDeactivationTime->isEarlier($lastCronArchiveTime) ? $lastDeactivationTime : $lastCronArchiveTime;
```

Comment:
Assuming that the last archive run time is smaller than the deactivation date, wouldn't it make sense to simply skip calling `scheduleReArchiving`. As no archives should have been built for any newer dates yet, they should be build with the next archiving process nevertheless, shouldn't they?

Reply:
last archive run time is not really so reliable. AFAIK it gets updated even if only few archives finish etc but doesn't mean they all finished. I reckon it's fine in worst case to rather rearchive a few days of reports

Reply:
they wouldn't because I don't think core:archive will create invalidations for them, but that's another problem I think... @tsteur if core:archive hasn't run for a few days, should we automatically create invalidations for those days? We'd check the last run time and if it's more than just today/yesterday, add more invalidations.